public/electron.js

```diff
@@ -1,96 +1,132 @@
+            )
+
+            reject(err)
+            return err
+          }
+
+          // Reconnect after 3 seconds
+          return 3000
         }
+      })
+
+      client.once('connect', () => {
+        // Rewrite AOF every minute
+        setInterval(() => {
+          client.bgrewriteaof()
+        }, 60000)
+
+        resolve()
+      })
+    })
+  }
 
-        // Reconnect after 3 seconds
-        return 3000
+  function initialize(queues) {
+    mainWindow = new BrowserWindow({
+      width: 1600,
+      height: 900,
+      icon: path.join(__dirname, '../src/assets/logos/128x128.png'),
+      show: false,
+      webPreferences: {
+        contextIsolation: false,
+        nodeIntegration: false,
+        preload: path.join(__dirname, 'preload.js')
       }
     })
 
-    client.once('connect', () => {
-      // Rewrite AOF every minute
-      setInterval(() => {
-        client.bgrewriteaof()
-      }, 60000)
+    mainWindow.loadURL(process.env.ELECTRON_DEV ?
+      'http://localhost:3000' :
+      `file://${path.join(__dirname, '../build/index.html')}`
+    )
 
-      resolve()
-    })
-  })
-}
-
-function initialize(queues) {
-  mainWindow = new BrowserWindow({
-    width: 1600,
-    height: 900,
-    icon: path.join(__dirname, '../src/assets/logos/128x128.png'),
-    show: false,
-    webPreferences: {
-      contextIsolation: false,
-      nodeIntegration: false,
-      preload: path.join(__dirname, 'preload.js')
+    mainWindow.webContents.on('did-finish-load', () => mainWindow.show())
+    // mainWindow.webContents.openDevTools()
+    if (process.env.PUBLISH50_DEV_TOOLS) {
+      mainWindow.webContents.openDevTools()
     }
-  })
+    mainWindow.on('closed', () => {
+      mainWindow = null
+    })
+  }
 
-  mainWindow.loadURL(process.env.ELECTRON_DEV ?
-    'http://localhost:3000' :
-    `file://${path.join(__dirname, '../build/index.html')}`
-  )
+  globalShortcut.register('CommandOrControl+Q', () => {
 
-  mainWindow.webContents.on('did-finish-load', () => mainWindow.show())
-  // mainWindow.webContents.openDevTools()
-  mainWindow.on('closed', () => {
-    mainWindow = null
-  })
-}
-
-app.on('ready', async () => {
+    if (isBoxOpen) {
+      return
+    }
+    else {
+      isBoxOpen = true
+    }
 
-  globalShortcut.register('CommandOrControl+Q', () => app.quit())
+    // Prompt user before closing the application if there are active jobs
+    if (Object.keys(queues['queues']).some((qname) => {
+      return Object.keys(queues['queues'][qname]['childPool'].retained).length > 0
+    })) {
+      dialog.showMessageBox(
+        {
+          type: 'question',
+          buttons: ['Cancel', 'Quit'],
+          message: `Looks like publish50 is currently running some tasks. Quitting publish50 will abort all running tasks. Are you sure you want to quit?`
+        },
+        (selectedIndex) => {
+          if (selectedIndex === 1) {
+            app.quit()
+          }
+          else {
+            isBoxOpen = false
```

Comment:
Since the box closes after clicking any button, it's probably a better idea to reset the flag at the top of this callback.